org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/configuration/DefaultConfigurationManager.java

```diff
@@ -159,6 +162,40 @@ public void deleteProjectConfiguration(IProject project) {
     @Override
     public RunConfiguration loadRunConfiguration(ILaunchConfiguration launchConfiguration) {
         GradleRunConfigurationAttributes attributes = GradleRunConfigurationAttributes.from(launchConfiguration);
+        ProjectConfiguration projectConfiguration = loadProjectConfiguration(attributes);
+        RunConfigurationProperties runConfigProperties = new RunConfigurationProperties(attributes.getTasks(),
+                  attributes.getGradleDistribution(),
+                  attributes.getGradleUserHome(),
+                  attributes.getJavaHome(),
+                  attributes.getJvmArguments(),
+                  attributes.getArguments(),
+                  attributes.isShowConsoleView(),
+                  attributes.isShowExecutionView(),
+                  attributes.isOverrideBuildSettings(),
+                  attributes.isBuildScansEnabled(),
+                  attributes.isOffline());
+        return new DefaultRunConfiguration(projectConfiguration, runConfigProperties);
+    }
+
+    @Override
+    public TestRunConfiguration loadTestRunConfiguration(ILaunchConfiguration launchConfiguration) {
```

Comment:
I think I get what's going on here... Before this change, we had a launch configuration for Gradle tasks that could be customized in all sorts of ways and since test execution does something very similar to task execution, you may need that same configuration.

Would it make more sense for some of this to be configured on a project level (console view, execution view, build scans, offline, gradle user home, etc) and only the things that are likely to be different (java, jvm arguments, arguments) be overrideable here? I think it's fine for this to be the same as task execution initially, but reducing the amount of configuration here seems easier for users to understand.  I can't think of a situation where you'd want to run your tests with a different version of Gradle.  I'm not sure that would even work in some cases because we use the project's gradle version to determine what features are available.

Reply:
The current state is in line with the UI. Each time you right-click on a test and choose `Debug As... > Gradle Test` a new launch configuration is persisted. By default, they all inherit the project settings, but you have the option to change it. I think it's useful to let the user configure all project properties for single test execution. They can play around with a single run configuration, without modifying the project config (which would have an affect on all other run configs).